server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -879,8 +881,20 @@ public Profilers getProfilers() {
      * Returns concurrent segment search status for the search context
      */
     @Override
-    public boolean isConcurrentSegmentSearchEnabled() {
-        return useConcurrentSearch;
+    public boolean shouldUseConcurrentSearch() {
```

Comment:
lets add a java doc stating that should be used after request parsing ?

Reply:
I wonder if we should just explicitly enforce this actually. For example `requestShouldUseConcurrentSearch` could be a ` final java.lang.Boolean` then we initialize it only once in `evaluateRequestShouldUseConcurrentSearch` and if it's `null` we throw an exception in this method. What do you think?

Reply:
Added an `assert` and javadoc in latest revision.